### PR TITLE
Add base_enterprise_url to reminder and nudge emails

### PR DESCRIPTION
**Pull request by raq929**
## Description

Passes a base_enterprise_url to ecommerce_worker, allowing email links to go to the learner portal.

## Testing instructions

Tested via sending reminders via the Admin portal.

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

File: ecommerce/extensions/api/serializers.py [hunk 1]
```diff
@@ -1918,11 +1919,13 @@ class Meta:
     code = serializers.CharField(required=True)
     email = serializers.EmailField(required=True)
     detail = serializers.CharField(read_only=True)
+    # base_enterprise_url = serializers.URLField(required=False)
```
Comment (edx-abolger) [inline ecommerce/extensions/api/serializers.py]: Remove commented out code

File: ecommerce/extensions/api/v2/views/enterprise.py [hunk 2]
```diff
@@ -837,13 +837,16 @@ def revoke(self, request, pk):  # pylint: disable=unused-argument
     def remind(self, request, pk):  # pylint: disable=unused-argument
         """
         Remind users of pending offer assignments by email.
+        coupons/<id>/remind
```
Comment (edx-abolger) [inline ecommerce/extensions/api/v2/views/enterprise.py:840]: I ❤️ easily searchable code urls

File: ecommerce/extensions/offer/tests/test_utils.py [hunk 3]
```diff
@@ -312,10 +312,64 @@ def test_send_assigned_offer_reminder_email(
...
+            'hi',
+            'bye',
+            'sender_alias',
+            {
+                'learner_email': 'johndoe@unknown.com',
+                'code': 'GIL7RUEOU7VHBH7Q',
+                'redeemed_offer_count': 0,
+                'total_offer_count': 1,
+                'code_expiration_date': '2018-12-19',
+                'base_enterprise_url': 'foo@bar.com'
```
Comment (edx-abolger) [inline ecommerce/extensions/offer/tests/test_utils.py]: tiny nit: can this be a formatted url? (it confused me to read it)

File: ecommerce/extensions/test/factories.py [hunk 4]
```diff
@@ -337,6 +337,7 @@ class CodeAssignmentNudgeEmailsFactory(factory.DjangoModelFactory):
     email_template = factory.SubFactory(CodeAssignmentNudgeEmailTemplatesFactory)
     user_email = factory.Sequence(lambda n: 'learner_%s@example.com' % n)
     email_date = datetime.now()
+    options = {'base_enterprise_url': 'foo@bar.com'}
```
Comment (edx-abolger) [inline ecommerce/extensions/test/factories.py]: URL?

**Review by edx-abolger — COMMENTED** [review]
Will wait on final PR

File: ecommerce/extensions/api/v2/views/enterprise.py (hunk 2 above)
Comment (raq929) [reply ecommerce/extensions/api/v2/views/enterprise.py:840]: omg the way this repo does URLs makes them really hard to trace so I keep adding them as I figure it out.

File: ecommerce/extensions/offer/tests/test_utils.py (hunk 3 above)
Comment (raq929) [reply ecommerce/extensions/offer/tests/test_utils.py]: OH yes it can, my bad.

**Review by edx-abolger — APPROVED** [review]

File: ecommerce/extensions/offer/models.py [hunk 5]
```diff
@@ -769,18 +769,23 @@ class CodeAssignmentNudgeEmails(TimeStampedModel):
...
+    def options_default():  # pylint: disable=no-method-argument
+        return {"base_enterprise_url": ''}
+
     email_template = models.ForeignKey('offer.CodeAssignmentNudgeEmailTemplates', on_delete=models.CASCADE)
     code = models.CharField(max_length=128, db_index=True)
     user_email = models.EmailField(db_index=True)
     email_date = models.DateTimeField()
     already_sent = models.BooleanField(help_text=_('Email has been sent.'), default=False)
     is_subscribed = models.BooleanField(help_text=_('This user should receive email'), default=True)
+    options = JSONField(default=options_default)
```
Comment (taliaedX) [inline ecommerce/extensions/offer/models.py:782]: actually,  why can't the default just be `default={"base_enterprise_url": ''}` I'm sure there's some Django nonsense reason, but I'm just curious

**Review by taliaedX — APPROVED** [review]
I am still curious about the oddities of the JSON field type, but still looks good.

File: ecommerce/extensions/offer/models.py (hunk 5 above)
Comment (raq929) [reply ecommerce/extensions/offer/models.py:782]: Yeah, I don't pretend to understand it, I just do what the docs tell me to do. 😛

Comment (taliaedX) [reply ecommerce/extensions/offer/models.py:782]: hmmmmm. I'm curious what the migration looks like without that, but I won't push it. As long as it works as defined it's not worth blocking.

**Pull request merged**